### PR TITLE
Feat: In dry run mode, don't delete any files

### PR DESCRIPTION
**Pull request by yahayaohinoyi**
## CONTEXT

We want the ability to run our commands and see what file would have been deleted

## HOW THIS WAS TESTES

`Unit test`
`Manually`

Screenshot 2024-09-16 at 18 13 56

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -8,3 +8,4 @@ clap = { version = "4.0", features = ["derive"] }
 walkdir = "2.3"
 anyhow = "1.0"
 tempfile = "3.12.0"
+colored = "2.0"
```
Comment (mubarak-ramadan) [inline Cargo.toml:11]: What does this do ?

File: src/arg.rs [hunk 2]
```diff
@@ -33,6 +34,14 @@ pub fn parse_args() -> Args {
                 .value_parser(clap::value_parser!(u64))
                 .help("Minimum size to clear (in bytes)"),
         )
+        .arg(
+            Arg::new("dry_run")
+                .short('n')
```
Comment (mubarak-ramadan) [inline src/arg.rs]: -n is not very intuitive for dry run.
Maybe -dr ? -r ?

File: src/main.rs [hunk 3]
```diff
@@ -8,14 +9,33 @@ fn main() -> Result<()> {
 
     // precedence: file types, then file size. In case they both show up together in the command
     // we'd do a refactor on here when we consider chaining
+
+    if args.dry_run {
```
Comment (mubarak-ramadan) [inline src/main.rs:13]: Nice

File: Cargo.toml (hunk 1 above)
Comment (yahayaohinoyi) [reply Cargo.toml:11]: fancy coloring. Don't want everything to be same grey color in the terminal

File: src/arg.rs (hunk 2 above)
Comment (yahayaohinoyi) [reply src/arg.rs]: yeah I though so too, was gonna go for -dr but it needs to be a character. d wasn't an option cause it's used. r feels like recursive but maybe better.  Will change

**Review by mubarak-ramadan — APPROVED** [review]

**Pull request merged**